cmd/kube-aws-approver/main.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"flag"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+
+	awsapprover "github.com/coreos/kubecsr/pkg/approver/aws"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"k8s.io/client-go/tools/record"
+)
+
+var (
+	kubeconfig  = flag.String("kubeconfig", "", "kubeconfig file with acces to cluster. (testing only)")
+	regionName  = flag.String("region-name", "", "When empty uses metadata service to extract.")
+	allowedASGs = flag.String("allowed-asgs", "", "Comma separated string of allowed ASGs")
+)
+
+func main() {
+	flag.Set("logtostderr", "true")
+	flag.Parse()
+
+	var config *rest.Config
+	var err error
+	if *kubeconfig != "" {
+		glog.V(4).Infof("Loading kube client config from path %q", *kubeconfig)
+		config, err = clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	} else {
+		glog.V(4).Infof("Using in-cluster kube client config")
+		config, err = rest.InClusterConfig()
+	}
+
+	aasgs := strings.Split(*allowedASGs, ",")
+	if len(aasgs) == 0 {
+		glog.Fatalf("error empty allowed asg list.")
+	}
+	arc := awsapprover.Config{
+		RegionName:  *regionName,
+		AllowedASGs: aasgs,
+	}
+
+	client := kubernetes.NewForConfigOrDie(config)
+	ar, err := awsapprover.New(client, arc)
+	if err != nil {
+		glog.Fatalf("error creating approver: %v", err)
+	}
+
+	lclient := kubernetes.NewForConfigOrDie(config)
+
+	id, err := os.Hostname()
+	if err != nil {
+		glog.Fatalf("error %v", err)
+	}
+	id = id + "_" + string(uuid.NewUUID())
+	recorder := record.
+		NewBroadcaster().
+		NewRecorder(runtime.NewScheme(), v1.EventSource{Component: "kube-aws-approver"})
+
+	rl, err := resourcelock.New(resourcelock.ConfigMapsResourceLock,
```

Comment:
Why leader election?

Reply:
Approver can be run as a deployment, that means multiple approvers can be active at the same time. Making sure only one acts on the CSRs ??